### PR TITLE
BAH-3175 | Added Previous Month option in Bahmni Reports pre-filter

### PR DESCRIPTION
**Pull request by deeptirawat1510**

### CODE REVIEW

**Comment by deeptirawat1510** [comment]
@gsluthra we have added test cases covering all the scenarios mentioned in the jira ticket, can you pls review this?

File: ui/test/unit/reports/controllers/reportsController.spec.js [hunk 1]
```diff
@@ -94,18 +102,42 @@ describe("ReportsController", function () {
...
         scope.setDefault('dateRangeType', 'reportsRequiringDateRange');
-        scope.setDefault('startDate', 'reportsRequiringDateRange');
-        scope.setDefault('stopDate', 'reportsRequiringDateRange');
 
         expect(_.keys(rootScope.default.reportsRequiringDateRange).length).toBe(3);
         expect((rootScope.default.reportsRequiringDateRange.dateRangeType).getDate()).toBe(new Date().getDate());
-        expect(rootScope.reportsRequiringDateRange[0].startDate).toBe(rootScope.default.reportsRequiringDateRange.startDate);
-        expect(rootScope.reportsRequiringDateRange[0].stopDate).toBe(rootScope.default.reportsRequiringDateRange.stopDate);
+        expect(rootScope.reportsRequiringDateRange[0].startDate.getDate()).toBe(new Date().getDate());
+        expect(rootScope.reportsRequiringDateRange[0].stopDate.getDate()).toBe(new Date().getDate());
```
Comment (gsluthra) [inline ui/test/unit/reports/controllers/reportsController.spec.js:112]: I didn't understand this assertion. The startdate and stop data is the same?

File: ui/test/unit/reports/controllers/reportsController.spec.js [hunk 2]
```diff
@@ -94,18 +102,42 @@ describe("ReportsController", function () {
...
+    });
+
+    const previousMonthTestCases = [
```
Comment (gsluthra) [inline ui/test/unit/reports/controllers/reportsController.spec.js:115]: Since all these scenarios have been added into a single datastructure, when this test fails, how does one know which use-case failed? Is there some log which prints the expected/calculated date?

File: ui/test/unit/reports/controllers/reportsController.spec.js [hunk 3]
```diff
@@ -94,18 +102,42 @@ describe("ReportsController", function () {
...
+        { currentDate: '1-Dec-2022', expectedStartDate: '1-Nov-2022', expectedStopDate: '30-Nov-2022' },
```
Comment (gsluthra) [inline ui/test/unit/reports/controllers/reportsController.spec.js:116]: Thanks for covering all the corner cases! appreciate this.

File: ui/test/unit/reports/controllers/reportsController.spec.js (hunk 2 above)
Comment (deeptirawat1510) [reply ui/test/unit/reports/controllers/reportsController.spec.js:115]: @gsluthra , when test case fails it would print the specific test case name for example- " should return previous date and month when ${currentDate}`" . Here current date would be the specific date.

Comment (gsluthra) [reply ui/test/unit/reports/controllers/reportsController.spec.js:115]: No, maybe this is sufficient.

File: ui/test/unit/reports/controllers/reportsController.spec.js (hunk 1 above)
Comment (gsluthra) [reply ui/test/unit/reports/controllers/reportsController.spec.js:112]: The test was for `today`, so makes sense. thanks!

**Review by gsluthra — APPROVED** [review]
